modules/core/plan/src/Plugin/FarmContext/PlanAssetFarmContext.php

```diff
@@ -0,0 +1,54 @@
+<?php
+
+namespace Drupal\plan\Plugin\FarmContext;
+
+use Drupal\farm_ui_context\Plugin\FarmContext\FarmContextBase;
+
+/**
+ * Provides context when an asset in a part of a plan.
+ *
+ * @todo Move to a module with a better dependency on the plan.asset field.
+ *
+ * @FarmContext(
+ *   id = "plan_asset_farm_context",
+ *   admin_label = @Translation("Plan asset farm context"),
+ *   context_definitions = {
+ *     "asset" = @ContextDefinition("entity:asset", label = @Translation("Asset")),
+ *   },
+ * )
+ */
+class PlanAssetFarmContext extends FarmContextBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getMessages(): array {
+    $messages = [];
+
+    /** @var \Drupal\asset\Entity\AssetInterface $asset */
+    if ($asset = $this->getContextValue('asset')) {
+
+      // Query for plans the asset might be a part of.
+      $plans = \Drupal::entityTypeManager()->getStorage('plan')->loadByProperties([
+        'asset' => $asset->id(),
```

Comment:
This is probably fine as-is, but we should confirm that this doesn't break on plans that do NOT include the default `asset` or `log` reference fields. It's possible for downstream plan types to omit those if they want to take their own approach (or use `plan_record` entities).

(I thought the `crop` and `grazing` plan modules both did that, but it seems that they are still inheriting the `asset` and `log` fields. That's a separate issue... those should be removed since they are not used by those plan types. I'll make note to follow up on that separately...)

Reply:
Ah interesting. As long as there is one plan bundle that has a `plan.asset` field I do not think this would cause an issue - but yes, if there were no plan bundles that had this `plan.asset` field, I can see how this might break. IIRC `loadByProperties` builds a simple entity API query behind the scenes which I do seem to remember failing when specified fields do not exist...